src/cli.ts

```diff
@@ -1,51 +1,537 @@
+        'regex of URLs to consider "internal"; all other URLs will be opened in an external browser.',
+      group: 'URL Handling Options',
+      type: 'string',
+    })
+    .option('proxy-rules', {
+      description:
+        'proxy rules; see https://www.electronjs.org/docs/api/session#sessetproxyconfig',
+      group: 'URL Handling Options',
+      type: 'string',
+    })
+    // Auth Options
+    .option('basic-auth-password', {
+      description: 'basic http(s) auth password',
+      group: 'Auth Options',
+      type: 'string',
+    })
+    .option('basic-auth-username', {
+      description: 'basic http(s) auth username',
+      group: 'Auth Options',
+      type: 'string',
+    })
+    // Graphics Options
+    .option('disable-gpu', {
+      default: false,
+      description: 'disable hardware acceleration',
+      group: 'Graphics Options',
+      type: 'boolean',
+    })
+    .option('enable-es3-apis', {
+      default: false,
+      description: 'force activation of WebGL 2.0',
+      group: 'Graphics Options',
+      type: 'boolean',
+    })
+    .option('ignore-gpu-blacklist', {
+      default: false,
+      description: 'force WebGL apps to work on unsupported GPUs',
+      group: 'Graphics Options',
+      type: 'boolean',
+    })
+    // Caching Options
+    .option('clear-cache', {
+      default: false,
+      description:
+        'prevent the application from preserving cache between launches',
+      group: 'Caching Options',
+      type: 'boolean',
+    })
+    .option('disk-cache-size', {
+      defaultDescription: 'chromium default',
+      description:
+        'forces the maximum disk space (in bytes) to be used by the disk cache',
+      group: 'Caching Options',
+      type: 'number',
+    })
+    // (In)Security Options
+    .option('disable-old-build-warning-yesiknowitisinsecure', {
+      default: false,
+      description:
+        'Disables warning when opening an app made with an old version of Nativefier. Nativefier uses the Chrome browser (through Electron), and it is dangerous to keep using an old version of it.)',
+      group: '(In)Security Options',
+      type: 'boolean',
+    })
+    .option('ignore-certificate', {
+      default: false,
+      description: 'ignore certificate-related errors',
+      group: '(In)Security Options',
+      type: 'boolean',
+    })
+    .option('insecure', {
+      default: false,
+      description: 'enable loading of insecure content',
+      group: '(In)Security Options',
+      type: 'boolean',
+    })
+    // Flash Options (DEPRECATED)
+    .option('flash', {
+      default: false,
+      deprecated: true,
+      description: 'enables Adobe Flash',
+      group: 'Flash Options (DEPRECATED)',
+      type: 'boolean',
+    })
+    .option('flash-path', {
+      deprecated: true,
+      description: 'path to Chrome flash plugin; find it in `chrome://plugins`',
+      group: 'Flash Options (DEPRECATED)',
+      normalize: true,
+      type: 'string',
+    })
```

Comment:
If `yargs` has a feature for this, can we actually hide these options (but keep honoring them if passed, by longtime users, or people reading API.md) ?

They're deprecated, after all, and consuming attention here.

Reply:
Yep! You can mark them as `hidden: true`.

I'll do that for these deprecated ones if you'd like.

Reply:
> Yep! You can mark them as `hidden: true`.
> 
> I'll do that for these deprecated ones if you'd like.

Great. Does that make sense, or would you rather keep them?

Reply:
Nope I think hiding them and marking them as deprecated in API.md is the way to go.

Reply:
> If `yargs` has a feature for this, can we actually hide these options (but keep honoring them if passed, by longtime users, or people reading API.md) ?

@TheCleric by the way, I laid out in [HACKING.md](https://github.com/nativefier/nativefier/blob/master/HACKING.md) the respect for backwards compatibility (said differently, limiting breaking changes) I'd like contributors to have. It was unsaid and only communicated implicitly until now, from now on it's written:

> Do your best to **limit breaking changes**. Only introduce breaking changes when necessary, when required by deps, or when not breaking would be unreasonable. When you can, support the old thing forever. For example, keep maintaining old flags; to "replace" an flag you want to replace with a better version, you should keep honoring the old flag, and massage it to pass parameters to the new flag, maybe using a wrapper/adapter.
>
> Yes, our code will get a tiny bit uglier than it could have been with a hard breaking change, but that would be to ignore our users. Introducing breaking changes willy nilly is a comfort to us developers, but is disrespectful to end users who must constantly bend to the flow of breaking changes pushed by _all their software_ who think it's "just one breaking change".